bin/git-jira-branch.rb

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env ruby
+# encoding: UTF-8
+#
+# git-jira-branch
+# 
+#   Suggest a branch name from the given JIRA issue ID
+# 
+#   Assumes the branches are named as:
+#   <team>/<branch-title>-<story-id>
+# 
+require 'rubygems'
+require 'optparse'
+require 'jira'
+require 'readline'
+require 'term/ansicolor'
+require 'git-whistles/app'
+
+class App < Git::Whistles::App
+
+  def initialize
+    super
+  end
+
+  def main(args)
+    super
+    parse_args!(args)
+
+    if args.count < 1
+      show_and_exit usage
+    end
+
+    story_id = args[0]
+
+    # get PT info
+    issue = get_jira_info(story_id)
+
+    branch_name_suggested = "#{story_id.upcase}-#{issue.fields['summary'].downcase}"
+    branch_name_suggested.gsub!(/[^\w\d\/]/, '-').gsub!(/-+/, '-')
+
+    puts 'The suggested branch name is: ' << Term::ANSIColor.yellow(branch_name_suggested)
+    puts '    Press ENTER if you agree'
+    puts ' or Write any other name and press ENTER'
+    puts ' or Press CTRL-D to cancel'
+
+    branch_name = Readline.readline '  > ', false
+
+    if branch_name.nil?
+      log.warn "\nCancelled by user"
+      exit 2
+    end
+
+    puts 'Would you like to transition the issue?'
+    issue.transitions.all.each do |t|
+      puts "#{t.id} - #{t.name}"
+    end
+    puts 'Type the number or enter to continue'
+    transition = Readline.readline '  > ', false
+
+    branch_name = branch_name.empty? ? branch_name_suggested : branch_name
+
+    # create branch
+    `cd #{ENV['PWD']} && git checkout -b #{branch_name}`
+
+    # comment and start on PT
+    # issue.notes.create :text => "Created branch #{branch_name}"
+    # issue.update :current_state => 'started'
+
+    if transition
+      # Seems we need to do this in two updates.  JIRA complained if we tried to transition and update fields.
+      attrs = {
```

Comment:
I'm not sure if this is your style or not, but indentation doesn't seem consistent between variable assignments?